build/cmake/global_compiler_flags.cmake

```
@@ -139,6 +139,14 @@ if(IS_MSVC)
     add_compile_definitions(UNICODE _UNICODE)
 endif()
 
+# Ensure that debug WASM builds include additional WASM-specific linker flags 
+if (IS_WASM AND CMAKE_BUILD_TYPE STREQUAL "Debug")
+    string(APPEND CMAKE_EXE_LINKER_FLAGS " -sASSERTIONS=2 -sSAFE_HEAP -sDEMANGLE_SUPPORT")
```

```suggestion
    string(APPEND CMAKE_EXE_LINKER_FLAGS " -sASSERTIONS=2 -sDEMANGLE_SUPPORT")
```

We do some custom alignment work if I heard Chris right this morning. Explains why I had to take this out to even boot the debug wasm.